pl_bolts/models/rl/advantage_actor_critic_model.py

```diff
@@ -0,0 +1,313 @@
+    <https://arxiv.org/abs/1602.01783v2>`_
+
+    Paper Authors: Volodymyr Mnih, Adrià Puigdomènech Badia, et al.
+
+    Model implemented by:
+
+        - `Jason Wang <https://github.com/blahBlahhhJ>`_
+
+    Example:
+        >>> from pl_bolts.models.rl.advantage_actor_critic_model import AdvantageActorCritic
+        ...
+        >>> model = AdvantageActorCritic("CartPole-v0")
+    """
+
+    def __init__(
+        self,
+        env: str,
+        gamma: float = 0.99,
+        lr: float = 0.001,
+        batch_size: int = 32,
+        n_steps: int = 10,
+        avg_reward_len: int = 100,
+        entropy_beta: float = 0.01,
+        critic_beta: float = 0.5,
+        epoch_len: int = 1000,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Args:
+            env: gym environment tag
+            gamma: discount factor
+            lr: learning rate
+            batch_size: size of minibatch pulled from the DataLoader
+            batch_episodes: how many episodes to rollout for each batch of training
+            avg_reward_len: how many episodes to take into account when calculating the avg reward
+            entropy_beta: dictates the level of entropy per batch
+            critic_beta: dictates the level of critic loss per batch
+            epoch_len: how many batches before pseudo epoch
+        """
+        super().__init__()
+
+        if not _GYM_AVAILABLE:  # pragma: no cover
+            raise ModuleNotFoundError("This Module requires gym environment which is not installed yet.")
+
+        # Hyperparameters
+        self.save_hyperparameters()
+        self.batches_per_epoch = batch_size * epoch_len
+
+        # Model components
+        self.env = gym.make(env)
+        self.net = ActorCriticMLP(self.env.observation_space.shape, self.env.action_space.n)
+        self.agent = ActorCriticAgent(self.net)
+
+        # Tracking metrics
+        self.total_rewards = [0]
+        self.episode_reward = 0
+        self.done_episodes = 0
+        self.avg_rewards = 0
+        self.avg_reward_len = avg_reward_len
+        self.eps = np.finfo(np.float32).eps.item()
+        self.batch_states = []
+        self.batch_actions = []
+        self.batch_rewards = []
+        self.batch_masks = []
+
+        self.state = self.env.reset()
+
+    def forward(self, x: torch.Tensor) -> Tuple[torch.Tensor]:
+        """
+        Passes in a state x through the network and gets the log prob of each action
+        and the value for the state as an output
+
+        Args:
+            x: environment state
+
+        Returns:
+            action log probabilities, values
+        """
+        if not isinstance(x, list):
+            x = [x]
+
+        if not isinstance(x, torch.Tensor):
+            x = torch.tensor(x, device=self.device)
+
+        logprobs, values = self.net(x)
+        return logprobs, values
+
+    def train_batch(self) -> Tuple[List[torch.Tensor], List[torch.Tensor], List[torch.Tensor]]:
```

Comment:
The typing here seems conflicting with the docstring below. Would you mind fixing it?

Reply:
This method is actually a generator. The doctoring's note is the type for the DataLoader (stacking all the yield). I changed the typing to the actual type that is YIELDed from this generator (Tuple[np.ndarray, int, torch.Tensor]). Not sure if it's ok (not sure how typing works for generators)

Reply:
Thanks for the update. For the record, I used `Iterator` as mentioned in the mypy docs: https://mypy.readthedocs.io/en/stable/kinds_of_types.html#generators.